### PR TITLE
Sanitize host before calling pm

### PR DESCRIPTION
**Pull request by blasten**
This is required before calling the package manager.

### CODE REVIEW

**Review by iskakaushik — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/fuchsia/fuchsia_pm.dart [hunk 1]
```diff
@@ -109,13 +109,14 @@ class FuchsiaPM {
...
     }
+    host = host.replaceAll('%', '%25');
     final List<String> command = <String>[
       fuchsiaArtifacts.pm.path,
       'serve',
       '-repo',
       repoPath,
       '-l',
-      '$host:$port',
+      '[$host]:$port',
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/fuchsia/fuchsia_pm.dart]: Is this guaranteed to be ipv6?

Comment (blasten) [reply packages/flutter_tools/lib/src/fuchsia/fuchsia_pm.dart]: Hmm it might need that check. Good catch

Comment (blasten) [reply packages/flutter_tools/lib/src/fuchsia/fuchsia_pm.dart]: Done! PTAL

**Review by jonahwilliams — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/fuchsia/fuchsia_device.dart [hunk 2]
```diff
@@ -513,12 +514,7 @@ class FuchsiaDevice extends Device {
   void clearLogs() {}
 
   bool get ipv6 {
-    try {
-      Uri.parseIPv6Address(id);
-      return true;
-    } on FormatException {
-      return false;
-    }
+    return isIPv6Address(id);
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/fuchsia/fuchsia_device.dart]: nit:
```dart
bool _ipv6;
bool get ipv6 => _ipv6 ??= isIPv6Address(id);
```

File: packages/flutter_tools/lib/src/fuchsia/fuchsia_utils.dart [hunk 3]
```diff
@@ -0,0 +1,13 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Returns [true] if [address] is an IPv6 address.
+bool isIPv6Address(String address) {
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/fuchsia/fuchsia_utils.dart]: Consider moving this to `base/net.dart`.

**Review by zanderso — COMMENTED** [review]
DBC

**Pull request merged**